File: azkry_android/app/src/main/kotlin/com/azkry/app/features/pages/viewmodels/PagesViewModel.kt
package com.azkry.app.features.pages.viewmodels

import androidx.lifecycle.ViewModel
import com.azkry.app.features.pages.models.AzkryPage
import com.azkry.app.features.pages.models.PageKey
import com.azkry.app.features.pages.models.PageSection
import com.azkry.app.features.pages.services.PagesService
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

/** Backs both the صفحات hub list and the individual topic pages. */
@HiltViewModel
class PagesViewModel @Inject constructor(
    private val pagesService: PagesService,
) : ViewModel() {
    val pages: List<AzkryPage> = pagesService.pages()

    val sayyidIstighfar: PageSection = pagesService.sayyidIstighfar()

    fun page(key: PageKey): AzkryPage? = pagesService.page(key)
}

File: azkary_ios/Azkry/Features/Pages/ViewModels/PagesViewModel.swift
import Observation

@MainActor
@Observable
final class PagesViewModel {
    private(set) var pages: [AzkryPage] = []
    private(set) var sayyidIstighfar: PageSection?
    private(set) var selectedPage: AzkryPage?
    private let service: any PagesServiceProtocol

    init(service: any PagesServiceProtocol = StaticPagesService()) {
        self.service = service
    }

    func load() {
        pages = service.pages()
        sayyidIstighfar = service.sayyidIstighfar()
    }

    func load(key: PageKey) {
        selectedPage = service.page(key: key)
    }
}
